schema/gnosis_protocol/view_trades.sql

```diff
@@ -16,72 +16,75 @@ JOIN gnosis_protocol."BatchExchange_call_submitSolution" solution
+    trades.evt_index AS evt_index_trades,
+    solution.evt_index AS evt_index_solution,
+    -- dates & block info
+    solution.evt_block_number,
+    solution.evt_block_time as block_time,
+    to_timestamp((floor(extract(epoch from solution.evt_block_time) / 300)) * 300) AS trade_date,
+    -- sell token
+    trades."sellToken" as sell_token_id,
     sell_token."token" as sell_token,
     sell_token."symbol" as sell_token_symbol,
     sell_token."decimals" as sell_token_decimals,
-    trades.executed_sell_amount as sell_amount_atoms,
-    trades.executed_sell_amount / 10^(sell_token.decimals) as sell_amount,
-    buy_token.token as buy_token,
+    -- sell amounts
+    trades."executedSellAmount" as sell_amount_atoms,
+    trades."executedSellAmount" / 10^(sell_token.decimals) as sell_amount,
+    -- buy token
+    trades."buyToken" as buy_token_id,
+    buy_token.token as buy_token,    
     buy_token.symbol as buy_token_symbol,
     buy_token.decimals as buy_token_decimals,
-    trades.executed_buy_amount as buy_amount_atoms,
-    trades.executed_buy_amount / 10^(buy_token.decimals) as buy_amount,
-    ((trades.executed_buy_amount / 10^(buy_token.decimals)) / (trades.executed_sell_amount / 10^(sell_token.decimals))) as price,
-    trades.tx_hash,
-    CONCAT('https://etherscan.io/tx/','0x', ENCODE(trades.tx_hash, 'hex')) as solution_tx_link,    
-    CONCAT('0x', ENCODE(trades."owner", 'hex')) as trader
+    -- buy amounts
+    trades."executedBuyAmount" as buy_amount_atoms,
+    trades."executedBuyAmount" / 10^(buy_token.decimals) as buy_amount,
+    -- Tx and block info
+    trades.evt_block_number as block_number,
+    trades.evt_tx_hash as tx_hash
+FROM gnosis_protocol."BatchExchange_evt_Trade" trades
+JOIN gnosis_protocol."BatchExchange_evt_SolutionSubmission" solution
+    ON solution.evt_tx_hash=trades.evt_tx_hash
+JOIN gnosis_protocol.view_tokens buy_token
+    ON trades."buyToken" = buy_token.token_id
+JOIN gnosis_protocol.view_tokens sell_token
+    ON trades."sellToken" = sell_token.token_id    
+)
+SELECT
+	trades.*,
+	reverts.block_time as revert_time
 FROM trades
 LEFT OUTER JOIN reverts
-    ON trades.owner = reverts.owner
+    ON trades.trader_hex = reverts.owner
     AND trades.order_id = reverts.order_id
     AND trades.trade_sub_id = reverts.trade_sub_id
     AND trades.batch_id = reverts.batch_id
-JOIN gnosis_protocol.view_tokens buy_token
-    ON trades.buy_token_id = buy_token.token_id
-JOIN gnosis_protocol.view_tokens sell_token
-    ON trades.sell_token_id = sell_token.token_id
+-- WHERE trades.batch_id = 5267142
 ORDER BY 
     trades.batch_id,
-    trades.owner,
+    trades.trader_hex,
     trades.order_id,
     trade_sub_id;
 
 
+
+
 CREATE UNIQUE INDEX IF NOT EXISTS view_trades_id ON gnosis_protocol.view_trades (batch_id, trader_hex, order_id, trade_sub_id);
-CREATE INDEX view_trades_1 ON gnosis_protocol.view_trades (batch_id);
+CREATE INDEX view_trades_1 ON gnosis_protocol.view_trades (trade_date);
```

Comment:
Postgres complains trade_date does not exist here.

Reply:
Strange, I see the field returned by the query:
![image](https://user-images.githubusercontent.com/2352112/81645730-60b17000-942a-11ea-988e-bdbfce597dcf.png)

Could it be that we need to use `DROP INDEX view_trades_1` first, since the index was previously defined?

Reply:
Yeah, I was surprised to, but will try now.
(WOW intellij does not like those bytearrays!)

Reply:
This was just a flaw in my testing setup (the index referenced the old view)